neqo-http3/src/connection.rs

```diff
@@ -323,6 +323,25 @@ impl<T: Http3Transaction> Http3Connection<T> {
         }
     }
 
+    pub fn is_critical_stream(&self, stream_id: u64) -> bool {
+        self.qpack_encoder
+            .local_stream_id()
+            .iter()
+            .chain(
+                self.qpack_encoder.remote_stream_id().iter().chain(
+                    self.qpack_decoder.local_stream_id().iter().chain(
+                        self.qpack_decoder.remote_stream_id().iter().chain(
+                            self.control_stream_local
+                                .stream_id()
+                                .iter()
+                                .chain(self.control_stream_remote.stream_id().iter()),
+                        ),
+                    ),
+                ),
+            )
```

Comment:
Rather than nesting, how about:

```suggestion
            .chain(self.qpack_encoder.remote_stream_id().iter())
            .chain(self.qpack_decoder.local_stream_id().iter())
            .chain(self.qpack_decoder.remote_stream_id().iter())
            .chain(self.control_stream_local.stream_id().iter())
            .chain(self.control_stream_remote.stream_id().iter())
```